### PR TITLE
Newly added Automated/semi-automated tests. Jan 31

### PR DESCRIPTION
**Pull request by manjunath-grl**
Newly added Media cluster automated tests:
```
TC_APPOBSERVER_1_13
TC_APPOBSERVER_13_2
TC_MEDIAPLAYBACK_6_8
TC_MEDIAPLAYBACK_6_9
```
ICDM semi-automated test:
`TC_ICDM_3_4`

Device Energy Management Mode tests:
```
TC_DEMM_1_1
TC_DEMM_2_1
```
Energy EVSE tests:
```
TC_EEVSE_1_1
TC_EEVSE_2_1
```
Refrigerator Alarm test:
`TC_REFALM_2_2`

Energy EVSE Mode tests:
```
TC_EEVSEM_1_1
TC_EEVSEM_2_1
TC_EEVSEM_3_1
TC_EEVSEM_3_2
TC_EEVSEM_3_3
```
Manual tests:
```
TC_DEMM_1_2
TC_EEVSEM_1_2
```

Added auto generated files

Tested
Execution logs:
[exe_logs.txt](https://github.com/project-chip/connectedhomeip/files/14111884/exe_logs.txt)

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_APPOBSERVER_1_13.yaml [hunk 1]
```diff
@@ -22,73 +22,73 @@ PICS:
...
-          [1702624983.368044][5900:5902] CHIP:TOO: Endpoint: 1 Cluster: 0x0000_0510 Attribute 0x0000_FFFB DataVersion: 2272699478
-          [1702624983.368485][5900:5902] CHIP:TOO:   AttributeList: 5 entries
-          [1702624983.368564][5900:5902] CHIP:TOO:     [1]: 65528
-          [1702624983.368624][5900:5902] CHIP:TOO:     [2]: 65529
-          [1702624983.368683][5900:5902] CHIP:TOO:     [3]: 65531
-          [1702624983.368741][5900:5902] CHIP:TOO:     [4]: 65532
-          [1702624983.368798][5900:5902] CHIP:TOO:     [5]: 65533
-          [1702624983.369236][5900:5902] CHIP:EM: <<< [E:55491i S:35892 M:103959395 (Ack:145764997)] (S) Msg TX to 1:0000000000000001 [EDAD] [UDP:[fe80::e65f:1ff:fe49:ae1a%wlan0]:5640] --- Type 0000:10 (SecureChannel:StandaloneAck)
-      disabled: true
+      PICS: "!PICS_EVENT_LIST_ENABLED"
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_APPOBSERVER_1_13.yaml:63]: Contains only checks that the values are part of the list, not that values not in the list are excluded. So these two pics conditions can effectively be reduced to a single condition

**Comment by cecille** [comment]
Which ones of these are automated, and which are manual?

File: examples/darwin-framework-tool/templates/tests/ciTests.json [hunk 2]
```diff
@@ -21,6 +21,8 @@
         "Test_TC_APPLAUNCHER_3_7",
         "Test_TC_APPLAUNCHER_3_8",
         "Test_TC_APPLAUNCHER_3_9",
+        "TC_MEDIAPLAYBACK_6_8",
```
Comment (cecille) [inline examples/darwin-framework-tool/templates/tests/ciTests.json:24]: This is really a large number of tests that should be reviewed by a different set of people.Can we please start breaking these PRs up and getting them properly reviewed by the appropriate feature devs if they're not going to write their own scripts? The PR title really doesn't indicate to any of them that they should be reviewing.

@raju-apple, @akrithika27

File: src/app/tests/suites/certification/Test_TC_DEMM_1_2.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+PICS:
+    - DEMM.S
+
+config:
+    nodeId: 0x12344321
+    cluster: "Basic Information"
+    endpoint: 1
+
+tests:
+    - label: "Step 1: TH reads from the DUT the SupportedModes attribute"
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_DEMM_1_2.yaml:27]: This is in a PR that says automated/semi-automated, but this looks like a fully manual test.

File: src/app/tests/suites/certification/Test_TC_DEMM_2_1.yaml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+      command: "readAttribute"
+      attribute: "CurrentMode"
+      response:
+          saveAs: Step6_current_mode_dut
+          constraints:
+              type: int8u
+              minValue: 0
+              maxValue: 255
+
+    #Skipping test-7 because SDK is not enabled with this failure response
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_DEMM_2_1.yaml:109]: If this cannot be done with the SDK, how was this step verified?

File: src/app/tests/suites/certification/Test_TC_EEVSEM_1_1.yaml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+          value: 1
+          constraints:
+              type: int16u
+
+      #https://github.com/project-chip/connectedhomeip/issues/31599
+    - label: "Step 3: TH reads from the DUT the FeatureMap attribute"
+      verification: |
+          ./chip-tool energyevsemode read feature-map 1 1
+
+          On the TH(Chip-tool) Log, Verify featureMap value is 0 and below is the sample log provided for the raspi platform:
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_EEVSEM_1_1.yaml:49]: Why does this need to be done manually?

File: src/app/tests/suites/certification/Test_TC_EEVSEM_1_2.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+PICS:
+    - EEVSEM.S
+
+config:
+    nodeId: 0x12344321
+    cluster: "Basic Information"
+    endpoint: 1
+
+tests:
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_EEVSEM_1_2.yaml:26]: This test appears to be fully manual.

File: src/app/tests/suites/certification/Test_TC_EEVSEM_2_1.yaml [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+          to PIXIT.EEVSEM.MODE_CHANGE_FAIL"
+      PICS:
+          EEVSEM.S.M.CAN_TEST_MODE_FAILURE && EEVSEM.S.C00.Rsp &&
+          PICS_SKIP_SAMPLE_APP
+      verification: |
+          Note : Please skip this step as SDK is not enabled with this failure response
+
+          ./chip-tool energyevsemode change-to-mode 2 1 1
+
+          Verify on TH(chip-tool) log, DUT responds contains a ChangeToModeResponse command with Status field is set to GenericFailure(0x02), InvalidInMode(0x03) , or in the MfgCodes (0x80 to 0xBF) range and StatusText field has a length between 1 and 64,  below is the sample log provided for the raspi platform:
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_EEVSEM_2_1.yaml:121]: With the current YAML support, i'm not sure this step will EVER be possible to automate. So why go down the YAML route here at all if we know we're going to end up at python?

**Pull request closed without merging**